### ISSUE TITLE
Error when invoking .set function

### ISSUE THREAD

**Issue by paradis-A** [appy-one/acebase#68]
Feel free to close this if irrelevant. Mentioning [possible fix](https://github.com/appy-one/acebase/pull/66).

[test] Index build /bounds/*/__search_field__ (fulltext) started
[test] Reading node "/" from address 0,0
[test] Node "/" being updated: adding 1 keys ("bounds"), updating 0 keys (), removing 0 keys ()
[test] Node "/bounds/ckzavyllh000b3p61u4fhbn8v/__search_field__" saved at address 0,17 - 1 addresses, 68 bytes written in 1 chunk(s)
[test] Node "/bounds/ckzavyllh000b3p61u4fhbn8v" saved at address 0,18 - 2 addresses, 144 bytes written in 1 chunk(s)
[test] Node "/bounds" saved at address 0,20 - 1 addresses, 38 bytes written in 1 chunk(s)
[test] Node "/" saved at address 0,0 - 1 addresses, 34 bytes written in 1 chunk(s)
[test] Reading node "/bounds/ckzavyllh000b3p61u4fhbn8v" from address 0,18
[test] Reading node "/bounds/ckzavyllh000b3p61u4fhbn8v" from address 0,18
[test] Reading node "/bounds/ckzavyllh000b3p61u4fhbn8v/__search_field__" from address 0,17
[test] Can't find event subscriptions to stop (path: "bounds/ckzavyllh000b3p61u4fhbn8v", event: (any), callback: undefined)
[test] Reading node "/bounds/ckzavyllh000b3p61u4fhbn8v/__search_field__" from address 0,17
[test] Indexed "/bounds/ckzavyllh000b3p61u4fhbn8v/__search_field__" value: 'ckzavyllh000b3p61u4fhbn8v,stockpileinboundtresttres' (object)
[test] done writing values to ./test.acebase/bounds-__search_field__.fulltext.idx.build
{ key: 'ckzavylli000c3p613rorwn7x', cost: 1500, quantity: 1 }
[test] Index build /bounds/*/materials/*/key started
[test] Index build /bounds/*/materials/*/__search_field__ (fulltext) started
[test] Reading node "/bounds/ckzavyllh000b3p61u4fhbn8v" from address 0,18
[test] done writing build file ./test.acebase/bounds-__search_field__.fulltext.idx.build
[test] Node "/bounds/ckzavyllh000b3p61u4fhbn8v" being updated: adding 1 keys ("materials"), updating 0 keys (), removing 0 keys ()
[test] Node "/bounds/ckzavyllh000b3p61u4fhbn8v/materials/ckzavylli000c3p613rorwn7x" saved at address 0,21 - 1 addresses, 113 bytes written in 1 chunk(s)
[test] Node "/bounds/ckzavyllh000b3p61u4fhbn8v/materials" saved at address 0,22 - 1 addresses, 38 bytes written in 1 chunk(s)
[test] Node "/bounds/ckzavyllh000b3p61u4fhbn8v" saved at address 0,18 - 2 addresses, 154 bytes written in 1 chunk(s)
C:\System\lenlen\lendb-server\node_modules\acebase\src\api-local.js:728
                    typeof oldValue === 'object' && Object.keys(oldValue).forEach(key => !seenKeys.includes(key) && seenKeys.push(key));
                                                           ^

TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at Object.childChangedCallback [as callback] (C:\System\lenlen\lendb-server\node_modules\acebase\src\api-local.js:728:60)
    at C:\System\lenlen\lendb-server\node_modules\acebase\src\storage.js:494:25
    at Array.forEach (<anonymous>)
    at Object.trigger (C:\System\lenlen\lendb-server\node_modules\acebase\src\storage.js:493:18)
    at C:\System\lenlen\lendb-server\node_modules\acebase\src\storage.js:1066:51
    at Array.forEach (<anonymous>)
    at triggerAllEvents (C:\System\lenlen\lendb-server\node_modules\acebase\src\storage.js:1012:14)
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
[nodemon] app crashed - waiting for file changes before starting...

**Comment by appy-one** [comment]
Thanks for your report and pull request, really appreciate it!

I do think this is a duplicate of #61 however

While your fix will definitely prevent the error, it does not fix the issue at hand. The main reason for this is because the event handler is expecting data to be delivered, but `notify_child_changed` event handlers do (and should) not get any data. As I've mentioned in the other issue, changing the `notify_child_changed` to `child_changed` subscription should be the right fix for now.

**Comment by appy-one** [comment]
Fix for this is published with acebase version 1.15.0

**Comment by paradis-A** [comment]
I confirm this is now working. Closing this now.
